WcaOnRails/app/webpacker/components/Panel/Board/RegionManager.jsx

```diff
@@ -45,50 +49,130 @@ function UserGroupVisibility({ userGroup, save, sync }) {
 
 export default function RegionManager() {
   const {
-    data, loading, error, sync,
+    data, loading: fetchLoading, error, sync,
   } = useLoadedData(fetchUserGroupsUrl('delegate_regions'));
   const { save, saving } = useSaveAction();
   const [openModalType, setOpenModalType] = React.useState();
   const [newRegion, setNewRegion] = React.useState(defaultRegion);
   const [saveError, setSaveError] = React.useState();
+  const [selectedGroup, setSelectedGroup] = React.useState();
+  const [newLeadDelegate, setNewLeadDelegate] = React.useState();
+  const [loading, setLoading] = React.useState(false);
+
+  React.useEffect(() => {
+    if (selectedGroup) {
+      setOpenModalType('newLeadDelegate');
+    }
+  }, [selectedGroup]);
```

Comment:
It's best to avoid side effects if possible. Just explicitly call `setOpenModalType('newLeadDelegate')` each time `setSelectedGroup` is set to a non-null value. Use a helper function if you want:
```
const selectedGroupAndShowModal = (group) -> {
  setSelectedGroup(group);
  setOpenModalType('newLeadDelegate');
}

Reply:
Makes sense, changed. Thank you Kevin.